tests/algorithms/test_centrality.py

```diff
@@ -195,3 +167,125 @@ def test_katz_centrality(edgelist1, edgelist8):
     }
     for n in c:
         assert np.allclose(c[n], expected_c[n])
+
+
+@pytest.mark.slow
+def test_h_eigenvector_centrality():
+    # test empty hypergraph
+    H = xgi.Hypergraph()
+    c = xgi.h_eigenvector_centrality(H)
+    assert c == dict()
+
+    # Test no edges
+    H.add_nodes_from([0, 1, 2])
+    hec = xgi.h_eigenvector_centrality(H)
+    for i in hec:
+        assert np.isnan(hec[i])
+
+    # test disconnected
+    H.add_edge([0, 1])
+    hec = xgi.h_eigenvector_centrality(H)
+    assert set(hec) == {0, 1, 2}
+    for i in hec:
+        assert np.isnan(hec[i])
+
+    H = xgi.sunflower(3, 1, 5)
+    c = xgi.h_eigenvector_centrality(H, max_iter=1000)
+    assert (
+        max([abs(c[0] / c[i + 1] - _ratio(3, 5, kind="HEC")) for i in range(12)]) < 1e-4
+    )
+
+    H = xgi.sunflower(5, 1, 7)
+    print(H.num_nodes)
+    c = xgi.h_eigenvector_centrality(H, max_iter=1000)
+    assert (
+        max([abs(c[0] / c[i + 1] - _ratio(5, 7, kind="HEC")) for i in range(29)]) < 1e-4
+    )
+
+    H = xgi.Hypergraph([[1, 2], [2, 3, 4]])
+    c = xgi.h_eigenvector_centrality(H)
+    true_c = {
+        1: 0.24458437592396465,
+        2: 0.3014043407819482,
+        3: 0.22700561916516002,
+        4: 0.22700566412892714,
+    }
+    for i in c:
+        assert np.allclose(c[i], true_c[i])
+
+
+@pytest.mark.slow
+def test_z_eigenvector_centrality():
+    # test empty hypergraph
+    H = xgi.Hypergraph()
+    c = xgi.z_eigenvector_centrality(H)
+    assert c == dict()
+
+    # Test no edges
+    H.add_nodes_from([0, 1, 2])
+    hec = xgi.z_eigenvector_centrality(H)
+    for i in hec:
+        assert np.isnan(hec[i])
+
+    # test disconnected
+    H.add_edge([0, 1])
+    hec = xgi.z_eigenvector_centrality(H)
+    assert set(hec) == {0, 1, 2}
+    for i in hec:
+        assert np.isnan(hec[i])
+
+    H = xgi.sunflower(3, 1, 5)
+    c = H.nodes.z_eigenvector_centrality(max_iter=1000).asdict()
+    assert (
+        max([abs(c[0] / c[i + 1] - _ratio(3, 5, kind="ZEC")) for i in range(12)]) < 1e-4
+    )
+
+    H = xgi.sunflower(5, 1, 7)
+    print(H.num_nodes)
+    c = xgi.z_eigenvector_centrality(H, max_iter=1000)
+    assert (
+        max([abs(c[0] / c[i + 1] - _ratio(5, 7, kind="ZEC")) for i in range(29)]) < 1e-4
+    )
+
+    H = xgi.Hypergraph([[1, 2], [2, 3, 4]])
+    c = xgi.z_eigenvector_centrality(H, max_iter=10000)
+    true_c = {
+        1: 0.45497398635982933,
+        2: 0.45900452108663403,
+        3: 0.04301074627676834,
+        4: 0.04301074627676829,
+    }
+    for i in c:
+        assert np.allclose(c[i], true_c[i])
+
+
+def _ratio(r, m, kind="CEC"):
```

Comment:
should this be in tests or in some utility file?

Reply:
Maybe? It really is only used to test centralities for the sunflower hypergraph, but since we only use it here, it didn't seem to make sense to move it.